src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -91,6 +96,9 @@ public Object instantiate() throws ConstructionException {
             logger.trace("Invoking constructor {} with arguments {}", ctor, ctorArgs);
             ctor.setAccessible(true);
             instance = ctor.newInstance(ctorArgs);
+            // ---  instantiating class pushed in MDC.-------------------------------------
+            mdcContext.put("org.grouplens.grapht.class", instance.getClass().toString());
```

Comment:
We want to store the value in the MDC earlier - before we start running any of the constructors. Really, before we start instantiating any of the constructor's arguments (so towards the beginning of this whole method).